lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -2693,10 +2693,16 @@ public function setAttributeOverride($fieldName, array $overrideMapping)
 
         $mapping = $this->fieldMappings[$fieldName];
 
-        //if (isset($mapping['inherited'])) {
-            // TODO: Enable this exception in 2.8
+        if (isset($mapping['inherited'])) {
+            Deprecation::trigger(
+                'doctrine/orm',
+                'https://github.com/doctrine/orm/pull/10470',
+                'Overrides are only allowed for fields or associations declared in mapped superclasses or traits, which is not the case for %s::%s. This is a misconfiguration and will be an error in Doctrine ORM 3.0.',
```

Comment:
Maybe mention where this was inherited from in the deprecation message. That information is available in `$mapping['inherited']`, correct?

Reply:
Yes, improved that.